### PR TITLE
ICU-22201 Add test to verify ICU-22043 on Java

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22201
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [X] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by yumaoka — APPROVED** [review]
LGTM

**Comment by markusicu** [comment]
@FrankYFTang please squash.

**Comment by FrankYFTang** [comment]
sorry, just back from sickness. Please approval again

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/calendar/CalendarRegressionTest.java [hunk 1]
```diff
@@ -2619,5 +2619,26 @@ public void TestTimeZoneInLocale20465() {
...
+        long startOfTime = -9007199254740992L;
+
+        cal.setGregorianChange(new Date(startOfTime));
+        cal.get(Calendar.ZONE_OFFSET);
+        cal.get(Calendar.DST_OFFSET);
+    }
+
+    @Test
+    public void TestAsiaManilaAfterSetGregorianChange22043() {
+        VerifyNoAssertWithSetGregorianChange("Asia/Malina");
```
Comment (richgillam) [inline icu4j/main/tests/core/src/com/ibm/icu/dev/test/calendar/CalendarRegressionTest.java:2637]: Do you mean `Asia/Manila` here?

Comment (FrankYFTang) [reply icu4j/main/tests/core/src/com/ibm/icu/dev/test/calendar/CalendarRegressionTest.java]: yes, thanks for catching that

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Review by richgillam — APPROVED** [review]
LGTM

**Pull request merged**